### ISSUE TITLE
intercept and handle calls that load additional scripts and other resources

### ISSUE THREAD

**Issue by mhempel-unl** [capricorn86/happy-dom#562]
Is it possible to intercept when happy-dom needs to load additional external resources? I would like to be able to intercept and control those events, decide whether to let it proceed, and possibly load the resource myself and provide the response back to happy-dom to continue processing, similar to jsdom's ResourceLoader capability. 
Thank you!

**Comment by azaslonov** [comment]
I have similar question. I would like just to render the HTML without loading the referenced resources. Is there an option to disable that?

**Comment by mhempel-unl** [comment]
> I have similar question. I would like just to render the HTML without loading the referenced resources. Is there an option to disable that?

yes, that is very similar to what i am trying to do as well. In JSDOM I can do this via a customresourceloader, to make sure nothing gets loaded from external sources. In other scenarios I can use this approach to control which resources I allow to be loaded. And a third use case for me is to use a specific method for resource loading, such as via libcurl with curl-impersonate. I would love to give happy-dom a try, but the lack of control over external resource loads is a dealbreaker.

**Comment by capricorn86** [comment]
Relates to #561

**Comment by 11Firefox11** [comment]
Are there any plans to implement this?
